packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts

```diff
@@ -93,15 +100,27 @@ export class Bootstrapper {
     if (cloudFormationExecutionPolicies.length === 0) {
       throw new Error('Please pass \'--cloudformation-execution-policies\' to specify deployment permissions. Try a managed policy of the form \'arn:aws:iam::aws:policy/<PolicyName>\'.');
     }
-    // Remind people what the current settings are
+
+    // * If an ARN is given, that ARN. Otherwise:
+    //   * '-' if customerKey = false
+    //   * '' if customerKey = true
+    //   * if customerKey is also not given
+    //     * undefined if we already had a value in place (reusing what we had)
+    //     * '-' if this is the first time we're deploying this stack (or upgrading from old to new bootstrap)
+    const kmsKeyId = params.kmsKeyId ??
+      (params.createCustomerMasterKey === true ? '' :
+        params.createCustomerMasterKey === false ? USE_AWS_MANAGED_KEY :
+          current.parameters.FileAssetsBucketKmsKeyId !== undefined ? undefined : USE_AWS_MANAGED_KEY);
```

Comment:
`~` Take or leave it, but removed a ternary, defined a new constant, and got this:

```suggestion
    const currentKmsKeyId = current.parameters.FileAssetsBucketKmsKeyId;
    const kmsKeyId = params.kmsKeyId ??
      (params.createCustomerMasterKey ? CREATE_CMK
        : (params.createCustomerMasterKey === false || currentKmsKeyId === undefined)
          ? USE_AWS_MANAGED_KEY : undefined);
```